components/brave_vpn/utils_win.cc

```diff
@@ -324,6 +356,44 @@ bool CreateEntry(const std::wstring& entry_name,
   return true;
 }
 
+CheckConnectionResult GetConnectionState(HRASCONN h_ras_conn) {
+  LPRASCONNSTATUS lp_ras_conn_status = NULL;
+  DWORD dw_ret = 0;
+
+  lp_ras_conn_status = (LPRASCONNSTATUS)HeapAlloc(
```

Comment:
This is definitely going to leak - `HeapAlloc` should always have a corresponding `HeapFree`. We can add the following directly after `RasGetConnectStatus`:

```cpp

RASCONNSTATE state = lp_ras_conn_status->rasconnstate;
HeapFree(GetProcessHeap(), 0, lp_ras_conn_status);
lp_ras_conn_status = NULL;

switch (state) {
```

Alternatively, we should be able to do this without the HeapAlloc; we can try changing the usage to:
```cpp
RASCONNSTATUS ras_conn_status ;

ZeroMemory(&ras_conn_status, sizeof(RASCONNSTATUS));
ras_conn_status->dwSize = sizeof(RASCONNSTATUS);

dw_ret = RasGetConnectStatus(h_ras_conn, &ras_conn_status);
// ..
switch (ras_conn_status.rasconnstate) {
```

This might re-use the same address and prevent memory fragmentation

Reply:
Oops, I missed free. Thanks for catching.
Updated w/o heap alloc.